framework/include/materials/MaterialProperty.h

```diff
@@ -330,7 +324,7 @@ template<typename P>
 PropertyValue *_init_helper(int size, PropertyValue * /*prop*/, const P*)
 {
   MaterialProperty<P> *copy = new MaterialProperty<P>;
-  copy->_value.resize(size);
+  copy->_value.resize(size, P{});
```

Comment:
This is necessary.  Without it, if ``P`` is a POD type then you can get uninitialized garbage.